src/webots/nodes/utils/WbTemplateManager.cpp

```diff
@@ -393,7 +393,7 @@ void WbTemplateManager::regenerateNode(WbNode *node, bool restarted) {
     foreach (WbField *parentParameter, previousParentRedirections) {
       foreach (WbField *newParam, newNode->parameters()) {
         if (parentParameter->name() == newParam->alias())
-          newParam->redirectTo(parentParameter);
+          newParam->redirectTo(parentParameter, true);
```

Comment:
Given that this a special case, I would probably block the signals just here without modifying the `WbField::redirectTo` function and adding a default argument.